services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/ApplicationConfig.java

```diff
@@ -323,6 +329,26 @@ public ObjectFactoryCreatingFactoryBean httpServerFactory() {
         return factory;
     }
 
+    /**
+     * Creates a new instance of an auth handler to provide basic authentication for the 
+     * Hono's Device Registry Management APIs.
+     * <p>
+     * This creates an instance of the {@link HonoBasicAuthHandler} with an auth provider of type
+     * {@link MongoAuth} to verify the credentials.
+     *
+     * @return The auth handler.
+     * @see <a href="https://vertx.io/docs/vertx-auth-mongo/java/">Mongo auth provider docs</a>
+     */
+    @Bean
+    @Scope("prototype")
+    @ConditionalOnProperty(name = "hono.registry.http.authenticationRequired", havingValue = "true", matchIfMissing = true)
+    public AuthHandler createAuthHandler() {
+        return new HonoBasicAuthHandler(
+                MongoAuth.create(mongoDBCallExecutor().getMongoClient(), new JsonObject()),
```

Comment:
I`d rather have a bean of type `MongoClient` being ceated by the config class which can then be used by multiple other beans. FMPOV there is no reason why we need to depend on `MongoDbCallExecutor` here ...

Reply:
I have refactored now to create a bean of type `MongoClient` instead of depending upon `MongoDbCallExecutor`.